pkg/controller/controller.go

```diff
@@ -374,8 +381,22 @@ func (s *Service) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest
 	}
 
 	if isMetroVolume {
-		// TODO configure Metro on volume
-		log.Warn("Configuring Metro on volume, not yet implemented.")
+		// Configure Metro on volume
+		volID := volumeResponse.VolumeId
+		log.Infof("Configuring Metro on volume %s", volID)
+
+		metroSession, err := arr.GetClient().ConfigureMetroVolume(ctx, volID, &gopowerstore.MetroConfig{
+			RemoteSystemID: remoteSystem.ID,
+		})
+		if err != nil {
+			if apiError, ok := err.(gopowerstore.APIError); ok && apiError.ReplicationSessionAlreadyCreated() { // idempotency check
+				log.Debugf("Metro has already been configured on volume %s", volID)
+			} else {
+				return nil, status.Errorf(codes.Internal, "can't configure metro on volume: %s", err.Error())
+			}
+		} else {
```

Comment:
could remove the `else` if you wanted to, here. Let me know and I can make the change so we don't conflict.